gmp: Add native libgmp to build native gdb

shahab-vahedi opened the pull request:
GDB 11.1+ requires libgmp to be built natively [1].  Therefore, add
libgmp building to the process of building native gdb.

[1] gdb: Make GMP a required dependency for building GDB
https://sourceware.org/git/?p=binutils-gdb.git;a=commit;h=1b4ac058f7d

Commit:
gmp: Add native libgmp to build native gdb

GDB 11.1+ requires libgmp to be built natively [1].  Therefore, add
libgmp building to the process of building native gdb.

[1] gdb: Make GMP a required dependency for building GDB
https://sourceware.org/git/?p=binutils-gdb.git;a=commit;h=1b4ac058f7d

anthony-kolesov commented:
I would recommend extracting the common code into arc-init.sh and calling as a function to avoid duplication (which is plentiful between uclibc and glibc files...). For example a `build_expat` function shared by all targets: https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/blob/arc-releases/arc-init.sh#L527

anthony-kolesov commented on build-glibc.sh:
```diff
@@ -488,6 +488,32 @@ if [ $DO_NATIVE_GDB = yes ]; then
+	--program-prefix= \
+	--enable-static
+    make_target building
+    make_target_ordered installing install DESTDIR=$SYSROOTDIR
```
Common code should be extracted into arc-init.sh , for example see a `build_expat` function shared by all targets: https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/blob/arc-releases/arc-init.sh#L527

shahab-vahedi replied:
Done. As a bonus, I've also moved the `ncurses` part.

Commit:
ncurses: Move the ncurses build into a common place

Both uclibc and glibc toolchain need to build ncurses for the
native GDB. The code for this process was duplicated over the
two variants. This commit shares the code among them by putting
it in a arc-init.sh as a common place.

anthony-kolesov commented:
After the current GNU release it makes sense to update to the latest ncurses version and remove the hack from 2016 as it will be not needed anymore.

shahab-vahedi commented:
It caught my attention too. I will tackle it at some other time. Thanks for the review!
